### DISCUSSION TITLE
Filter history by date range

### DISCUSSION THREAD

**Discussion by MidnightRetro** [OpenTracksApp/OpenTracks#1045] [Ideas]
It would be REALLY great to be able to **filter** the history by **date range**... With **totals**!
Like "How many kilometers have I jogged this month?" or "How many kilometers did I cycle last year?" Tap _filter, range_... BOOM, that easy!

**Comment by rgmf** [comment]
There are more info about it here: https://github.com/OpenTracksApp/OpenTracks/issues/1039

Also, there is an opened PR to filter selecting tracks in the list of tracks: https://github.com/OpenTracksApp/OpenTracks/pull/1040

Finally, a time ago we discussed about the needed of an analytics part but it's not the OpenTracks main goal.

Anyway, it seems feasible to add some type of filtering in the Aggregated Stats activity. Could you add it? Would you like to add it yourself? Do you have time/skills to do it?

**Comment by MidnightRetro** [comment]
Ooooh, I wish I could help! 😔
I have literally ZERO experience in coding.
But to be honest I'm thinking about taking a course or something. If you have some advice, I might learn how to do it myself. 😁

Anyway, it seems like there's quite a demand for this feature so I'd really appreciate if you could look into it. 😇

**Reply by rgmf** [reply]
Don't worry. I think it would be a great feature so I'll try to do it in the next weeks.

**Reply by MidnightRetro** [reply]
Aww, thanks!!
Yeah, I guess that's a quicker solution 😅

**Comment by dennisguse** [comment]
@thiswastheonlyavailableusername FYI: @rgmf implemented this feature and it will be in the next release

**Reply by MidnightRetro** [reply]
Thanks @rgmf

**Reply by rgmf** [reply]
@thiswastheonlyavailableusername you're welcome, and thank you for open this discussion. Many times are more important the idea that coding that idea ;)
